app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,35 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    # @post = Post.find(params[:author_id])
-    @posts = Post.where(author: @user).order(id: :asc).limit(2)
+    @posts = Post.where(author: @user).order(id: :asc)
   end
 
   def show
+    @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
     @comments = @post.comments
   end
+
+  def new
+    post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: } }
+    end
+  end
+
+  def create
+    post = Post.new(params.require(:post).permit(:author, :title, :text))
+    post.author = current_user
+    respond_to do |format|
+      format.html do
+        if post.save
+          flash[:success] = 'Post saved successfully'
+          redirect_to users_url
+        else
+          flash.now[:error] = 'Error: Post could not be saved'
+          render :new, status: 422
+        end
+      end
+    end
+  end
```

Comment:
Posts controller is perfect ✔

Reply:
Hi! @mateo951 Thanks for your review and your appreciation!